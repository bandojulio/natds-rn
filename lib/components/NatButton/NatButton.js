Object.defineProperty(exports,"__esModule",{value:true});exports.NatButton=undefined;var _jsxFileName='lib/components/NatButton/NatButton.js';var _react=require('react');var _react2=_interopRequireDefault(_react);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _ic_arrow_left_white=require('../../assets/icons/ic_arrow_left_white.png');var _ic_arrow_left_white2=_interopRequireDefault(_ic_arrow_left_white);

var _NatButton=require('./NatButton.styles');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}

var NatButton=exports.NatButton=function NatButton(_ref){var
onPress=_ref.onPress,type=_ref.type,label=_ref.label,noUpperCase=_ref.noUpperCase,iconLeft=_ref.iconLeft;return(

_react2.default.createElement(_NatButton.ButtonWrapper,{
accessible:true,
accessibilityRole:'button',
onPress:onPress,
type:type,
disabled:type==='disabled',__source:{fileName:_jsxFileName,lineNumber:10}},

iconLeft&&_react2.default.createElement(_NatButton.ButtonIcon,{source:iconLeft,__source:{fileName:_jsxFileName,lineNumber:17}}),
_react2.default.createElement(_NatButton.ButtonText,{
type:type,
accessibilityLabel:''+label,__source:{fileName:_jsxFileName,lineNumber:18}},

noUpperCase?label:label.toUpperCase()),

type==='card'&&_react2.default.createElement(_NatButton.ButtonIcon,{source:_ic_arrow_left_white2.default,__source:{fileName:_jsxFileName,lineNumber:24}})));};



NatButton.propTypes={
iconLeft:_propTypes2.default.string,
label:_propTypes2.default.string.isRequired,
noUpperCase:_propTypes2.default.bool,
onPress:_propTypes2.default.func.isRequired,
type:_propTypes2.default.string};


NatButton.defaultProps={
type:'default'};
//# sourceMappingURL=NatButton.js.map