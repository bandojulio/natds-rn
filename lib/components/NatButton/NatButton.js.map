{"version":3,"sources":["NatButton.js"],"names":["NatButton","onPress","type","label","noUpperCase","iconLeft","toUpperCase","icoArrowLeft","propTypes","PropTypes","string","isRequired","bool","func","defaultProps"],"mappings":"8IAAA,4B;AACA,qC;AACA,+E;;AAEA,6C;;AAEO,GAAMA,6BAAY,QAAZA,UAAY;AACvBC,OADuB,MACvBA,OADuB,CACdC,IADc,MACdA,IADc,CACRC,KADQ,MACRA,KADQ,CACDC,WADC,MACDA,WADC,CACYC,QADZ,MACYA,QADZ;;AAGrB,8BAAC,wBAAD;AACE,eADF;AAEE,kBAAkB,QAFpB;AAGE,QAASJ,OAHX;AAIE,KAAMC,IAJR;AAKE,SAAUA,OAAS,UALrB;;AAOGG,UAAY,8BAAC,qBAAD,EAAY,OAAQA,QAApB,iDAPf;AAQE,8BAAC,qBAAD;AACE,KAAMH,IADR;AAEE,sBAAuBC,KAFzB;;AAIGC,YAAcD,KAAd,CAAsBA,MAAMG,WAAN,EAJzB,CARF;;AAcGJ,OAAS,MAAT,EAAmB,8BAAC,qBAAD,EAAY,OAAQK,6BAApB,iDAdtB,CAHqB,GAAlB;;;;AAqBPP,UAAUQ,SAAV,CAAsB;AACpBH,SAAUI,oBAAUC,MADA;AAEpBP,MAAOM,oBAAUC,MAAV,CAAiBC,UAFJ;AAGpBP,YAAaK,oBAAUG,IAHH;AAIpBX,QAASQ,oBAAUI,IAAV,CAAeF,UAJJ;AAKpBT,KAAMO,oBAAUC,MALI,CAAtB;;;AAQAV,UAAUc,YAAV,CAAyB;AACvBZ,KAAM,SADiB,CAAzB","file":"NatButton.js","sourcesContent":["import React from 'react';\nimport PropTypes from 'prop-types';\nimport icoArrowLeft from '../../assets/icons/ic_arrow_left_white.png';\n\nimport { ButtonWrapper, ButtonText, ButtonIcon } from './NatButton.styles';\n\nexport const NatButton = ({\n  onPress, type, label, noUpperCase, iconLeft,\n}) => (\n    <ButtonWrapper\n      accessible\n      accessibilityRole=\"button\"\n      onPress={onPress}\n      type={type}\n      disabled={type === 'disabled'}\n    >\n      {iconLeft && <ButtonIcon source={iconLeft} />}\n      <ButtonText\n        type={type}\n        accessibilityLabel={`${label}`}\n      >\n        {noUpperCase ? label : label.toUpperCase()}\n      </ButtonText>\n      {type === 'card' && <ButtonIcon source={icoArrowLeft} />}\n    </ButtonWrapper>\n);\n\nNatButton.propTypes = {\n  iconLeft: PropTypes.string,\n  label: PropTypes.string.isRequired,\n  noUpperCase: PropTypes.bool,\n  onPress: PropTypes.func.isRequired,\n  type: PropTypes.string,\n};\n\nNatButton.defaultProps = {\n  type: 'default',\n};\n"]}