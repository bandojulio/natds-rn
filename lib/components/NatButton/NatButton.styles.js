import styled, { css } from 'styled-components/native';
import { robotoBold } from '../../assets/utils/fonts';
import colors from '../../assets/utils/colors';

const buttonStyles = {
  default: {
    backgroundColor: colors.primary.white,
    width: 328,
    height: 48,
    marginRight: 0,
  },
  short: {
    backgroundColor: colors.primary.white,
    width: 328,
    height: 40,
    marginRight: 0,
  },
  card: {
    backgroundColor: colors.primary.yellow,
    width: 128,
    height: 80,
    marginRight: 30,
  },
};

const textStyles = {
  default: { color: colors.primary.yellow },
  card: { color: colors.primary.white },
};

const getStylesFrom = styles => type => styles[type] || styles.default;

const getButtonStyles = getStylesFrom(buttonStyles);

const getTextStyles = getStylesFrom(textStyles);

export const ButtonWrapper = styled.TouchableOpacity`
  background-color: ${({ type }) => getButtonStyles(type).backgroundColor};
  height: ${({ type }) => getButtonStyles(type).height};
  width: ${({ type }) => getButtonStyles(type).width};
  margin-right: ${({ type }) => getButtonStyles(type).marginRight};
  border-radius: 6;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  ${props => props.secondary && css`
    background-color: none;
  `}
`;

export const ButtonText = styled.Text`
  color: ${({ type }) => getTextStyles(type).color};
  font-size: 14px;
  align-self: center;
  font-weight: bold;
  ${robotoBold};

  ${props => props.type === 'card' && css`
    width: 64px;
    text-align: center;
    line-height: 21px;
  `}
`;

export const ButtonIcon = styled.Image`
  height: 30;
  width: 30;
`;
