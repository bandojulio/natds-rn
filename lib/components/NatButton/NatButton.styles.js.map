{"version":3,"sources":["NatButton.styles.js"],"names":["buttonStyles","card","backgroundColor","colors","primary","yellow","color","white","height","width","default","disabled","bw","grayLight","grayDark","link","content","info","newDefault","grayDarker","newShort","outline","borderColor","borderWidth","short","getStylesFrom","styles","type","getButtonStyles","ButtonWrapper","styled","TouchableOpacity","ButtonText","Text","Platform","OS","sanfrancisco","bold","roboto","props","css","ButtonIcon","Image"],"mappings":"wrCAAA;AACA,gD;AACA;AACA,wD;;AAEA,GAAMA,cAAe;AACnBC,KAAM;AACJC,gBAAiBC,eAAOC,OAAP,CAAeC,MAD5B;AAEJC,MAAOH,eAAOC,OAAP,CAAeG,KAFlB;AAGJC,OAAQ,GAHJ;AAIJC,MAAO,GAJH,CADa;;AAOnBC,QAAS;AACPR,gBAAiBC,eAAOC,OAAP,CAAeC,MADzB;AAEPC,MAAOH,eAAOC,OAAP,CAAeG,KAFf;AAGPC,OAAQ,EAHD,CAPU;;AAYnBG,SAAU;AACRT,gBAAiBC,eAAOS,EAAP,CAAUC,SADnB;AAERP,MAAOH,eAAOS,EAAP,CAAUE,QAFT;AAGRN,OAAQ,EAHA,CAZS;;AAiBnBO,KAAM;AACJb,gBAAiB,aADb;AAEJI,MAAOH,eAAOa,OAAP,CAAeC,IAFlB;AAGJT,OAAQ,EAHJ,CAjBa;;AAsBnBU,WAAY;AACVhB,gBAAiBC,eAAOC,OAAP,CAAeC,MADtB;AAEVC,MAAOH,eAAOS,EAAP,CAAUO,UAFP;AAGVX,OAAQ,EAHE,CAtBO;;AA2BnBY,SAAU;AACRlB,gBAAiBC,eAAOC,OAAP,CAAeG,KADxB;AAERD,MAAOH,eAAOS,EAAP,CAAUO,UAFT;AAGRX,OAAQ,EAHA;AAIRC,MAAO,GAJC,CA3BS;;AAiCnBY,QAAS;AACPnB,gBAAiBC,eAAOC,OAAP,CAAeG,KADzB;AAEPe,YAAanB,eAAOC,OAAP,CAAeC,MAFrB;AAGPkB,YAAa,CAHN;AAIPjB,MAAOH,eAAOC,OAAP,CAAeC,MAJf;AAKPG,OAAQ,EALD,CAjCU;;AAwCnBgB,MAAO;AACLtB,gBAAiBC,eAAOC,OAAP,CAAeG,KAD3B;AAELD,MAAOH,eAAOC,OAAP,CAAeC,MAFjB;AAGLG,OAAQ,EAHH;AAILC,MAAO,GAJF,CAxCY,CAArB;;;;AAgDA,GAAMgB,eAAgB,QAAhBA,cAAgB,eAAU,sBAAQC,QAAOC,IAAP,GAAgBD,OAAOhB,OAA/B,EAAV,EAAtB;;AAEA,GAAMkB,iBAAkBH,cAAczB,YAAd,CAAxB;;AAEO,GAAM6B,qCAAgBC,iBAAOC,gBAAvB;AACS,kBAAGJ,KAAH,MAAGA,IAAH,OAAcC,iBAAgBD,IAAhB,EAAsBzB,eAApC,EADT;AAED,mBAAGyB,KAAH,OAAGA,IAAH,OAAcC,iBAAgBD,IAAhB,EAAsBnB,MAApC,EAFC;AAGF,mBAAGmB,KAAH,OAAGA,IAAH,OAAcC,iBAAgBD,IAAhB,EAAsBlB,KAAtB,EAA+B,MAA7C,EAHE;;;;;AAQK,mBAAGkB,KAAH,OAAGA,IAAH,OAAcC,iBAAgBD,IAAhB,EAAsBL,WAAtB,EAAqC,aAAnD,EARL;AASK,mBAAGK,KAAH,OAAGA,IAAH,OAAcC,iBAAgBD,IAAhB,EAAsBJ,WAAtB,EAAqC,CAAnD,EATL,CAAN;;;AAYA,GAAMS,+BAAaF,iBAAOG,IAApB;AACF,mBAAGN,KAAH,OAAGA,IAAH,OAAcC,iBAAgBD,IAAhB,EAAsBrB,KAApC,EADE;;;;;AAMM4B,sBAASC,EAAT,GAAgB,KAAjB,CAA0BC,oBAAaC,IAAvC,CAA8CC,cAAOD,IAN1D;;AAQT,sBAASE,OAAMZ,IAAN,GAAe,MAAf,KAAyBa,WAAzB,mBAAT,EARS,CAAN;;;;;;;AAeA,GAAMC,+BAAaX,iBAAOY,KAApB,kBAAN","file":"NatButton.styles.js","sourcesContent":["import { Platform } from 'react-native';\nimport styled, { css } from 'styled-components/native';\nimport { roboto, sanfrancisco } from '../../assets/theme/fonts/fonts';\nimport { colors } from '../../assets/theme/colors/colors';\n\nconst buttonStyles = {\n  card: {\n    backgroundColor: colors.primary.yellow,\n    color: colors.primary.white,\n    height: 128,\n    width: 128,\n  },\n  default: {\n    backgroundColor: colors.primary.yellow,\n    color: colors.primary.white,\n    height: 48,\n  },\n  disabled: {\n    backgroundColor: colors.bw.grayLight,\n    color: colors.bw.grayDark,\n    height: 48,\n  },\n  link: {\n    backgroundColor: 'transparent',\n    color: colors.content.info,\n    height: 48,\n  },\n  newDefault: {\n    backgroundColor: colors.primary.yellow,\n    color: colors.bw.grayDarker,\n    height: 48,\n  },\n  newShort: {\n    backgroundColor: colors.primary.white,\n    color: colors.bw.grayDarker,\n    height: 40,\n    width: 328,\n  },\n  outline: {\n    backgroundColor: colors.primary.white,\n    borderColor: colors.primary.yellow,\n    borderWidth: 2,\n    color: colors.primary.yellow,\n    height: 48,\n  },\n  short: {\n    backgroundColor: colors.primary.white,\n    color: colors.primary.yellow,\n    height: 40,\n    width: 328,\n  },\n};\n\nconst getStylesFrom = styles => type => styles[type] || styles.default;\n\nconst getButtonStyles = getStylesFrom(buttonStyles);\n\nexport const ButtonWrapper = styled.TouchableOpacity`\n  background-color: ${({ type }) => getButtonStyles(type).backgroundColor};\n  height: ${({ type }) => getButtonStyles(type).height};\n  width: ${({ type }) => getButtonStyles(type).width || '100%'};\n  border-radius: 6;\n  align-items: center;\n  justify-content: center;\n  flex-direction: row;\n  border-color: ${({ type }) => getButtonStyles(type).borderColor || 'transparent'};\n  border-width: ${({ type }) => getButtonStyles(type).borderWidth || 0};\n`;\n\nexport const ButtonText = styled.Text`\n  color: ${({ type }) => getButtonStyles(type).color};\n  font-size: 14px;\n  align-self: center;\n  font-weight: bold;\n  letter-spacing: 1;\n  font-family:  ${(Platform.OS === 'ios') ? sanfrancisco.bold : roboto.bold};\n\n  ${props => props.type === 'card' && css`\n    width: 64px;\n    text-align: center;\n    line-height: 21px;\n  `}\n`;\n\nexport const ButtonIcon = styled.Image`\n  height: 30;\n  width: 30;\n`;\n"]}