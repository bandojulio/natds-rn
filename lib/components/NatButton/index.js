Object.defineProperty(exports,"__esModule",{value:true});var _NatButton=require('./NatButton');Object.defineProperty(exports,'NatButton',{enumerable:true,get:function get(){return _NatButton.NatButton;}});
//# sourceMappingURL=index.js.map