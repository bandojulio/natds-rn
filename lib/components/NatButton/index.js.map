{"version":3,"sources":["index.js"],"names":["NatButton"],"mappings":"+LAASA,S","file":"index.js","sourcesContent":["export { NatButton } from './NatButton';\n"]}