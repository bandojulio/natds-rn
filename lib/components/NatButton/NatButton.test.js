import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components/native';

import { Text } from 'react-native';
import { NatButton } from './NatButton';

import { ButtonIcon } from './NatButton.styles';

describe('Button component default', () => {
  it('Should render button component ', () => {
    const button = renderer.create(
      <NatButton label="string" onPress={jest.fn()}>
        <Text>Oi</Text>
      </NatButton>,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

  describe('when is card type', () => {
    const props = {
      type: 'card',
      label: 'test',
      onPress: jest.fn(),
    };

    it('Should render button component ', () => {
      const button = renderer.create(<NatButton {...props} />).toJSON();
      expect(button).toMatchSnapshot();
    });

    it('should have an icon', () => {
      const wrapper = shallow(<NatButton {...props} />);
      expect(wrapper.find(ButtonIcon).length).toBe(1);
    });
  });

  describe('when is short type', () => {
    const props = {
      type: 'short',
      label: 'test',
      onPress: jest.fn(),
    };

    it('Should render short button component', () => {
      const button = renderer.create(<NatButton {...props} />).toJSON();
      expect(button).toMatchSnapshot();
    });

    it('should not have an icon', () => {
      const wrapper = shallow(<NatButton {...props} />);
      expect(wrapper.find(ButtonIcon).length).toBe(0);
    });
  });

  describe('when is outline type', () => {
    const props = {
      type: 'outline',
      label: 'test',
      onPress: jest.fn(),
    };

    it('Should render outline button component', () => {
      const button = renderer.create(<NatButton {...props} />).toJSON();
      expect(button).toMatchSnapshot();
    });

    it('should not have an icon', () => {
      const wrapper = shallow(<NatButton {...props} />);
      expect(wrapper.find(ButtonIcon).length).toBe(0);
    });
  });
});
