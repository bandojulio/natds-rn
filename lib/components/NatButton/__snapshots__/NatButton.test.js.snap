// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render button component  1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4AB34",
      "borderRadius": 6,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "marginRight": 0,
      "opacity": 1,
      "width": 328,
    }
  }
>
  <Text
    accessibilityLabel="string"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      ]
    }
    type="default"
  >
    STRING
  </Text>
</View>
`;

exports[`Button component when is card type Should render button component  1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4AB34",
      "borderRadius": 6,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "center",
      "marginRight": 30,
      "opacity": 1,
      "width": 128,
    }
  }
>
  <Text
    accessibilityLabel="test"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 22,
          "textAlign": "center",
          "width": 64,
        },
      ]
    }
    type="card"
  >
    TEST
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../lib/assets/icons/ic_arrow_left_white.png",
      }
    }
    style={
      Array [
        Object {
          "height": 30,
          "width": 30,
        },
      ]
    }
  />
</View>
`;

exports[`Button component when it has different type and it is secondary Should render button component  1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 6,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "marginRight": 0,
      "opacity": 1,
      "width": 328,
    }
  }
>
  <Text
    accessibilityLabel="test"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      ]
    }
    type="another"
  >
    TEST
  </Text>
</View>
`;
